File: scripts/debug_query.py
import sqlite3
import json

db_path = r'C:\Users\1104289\AppData\Roaming\meetingminutes\data\meetingminutes.db'

try:
    conn = sqlite3.connect(db_path)
    conn.row_factory = sqlite3.Row
    cursor = conn.cursor()

    print('--- ALL JOBS ---')
    cursor.execute('SELECT id, source_name, status, created_at FROM jobs')
    jobs = [dict(row) for row in cursor.fetchall()]
    print(json.dumps(jobs, indent=2, ensure_ascii=False))

    print('--- SETTINGS ---')
    cursor.execute('SELECT key, value FROM settings')
    settings = [dict(row) for row in cursor.fetchall()]
    print(json.dumps(settings, indent=2, ensure_ascii=False))

    print('--- SEARCHING FOR TARGET JOB ---')
    cursor.execute("SELECT * FROM jobs WHERE source_name LIKE '%2026-05-29%' OR source_name LIKE '%1.m4a%'")
    target_jobs = [dict(row) for row in cursor.fetchall()]
    if target_jobs:
        for job in target_jobs:
            print(f"Job Found: {job['id']} - {job['source_name']}")
            print(f"Status: {job['status']}")
            print(f"Transcript Length: {len(job['transcript_text'])}")
            print(f"Transcript Snippet:\n{job['transcript_text'][:500]}...\n")
            print(f"Summary JSON:\n{job['summary_json']}\n")
    else:
        print('No matching job found!')

except Exception as e:
    print('Error:', e)
finally:
    if 'conn' in locals():
        conn.close()

File: scripts/debug_query.js
const Database = require('better-sqlite3');
const db = new Database('C:\\Users\\1104289\\AppData\\Roaming\\meetingminutes\\data\\meetingminutes.db');

try {
  console.log('--- ALL JOBS ---');
  const jobs = db.prepare('SELECT id, source_name, status, created_at FROM jobs').all();
  console.log(JSON.stringify(jobs, null, 2));

  console.log('--- SETTINGS ---');
  const settings = db.prepare('SELECT key, value FROM settings').all();
  console.log(JSON.stringify(settings, null, 2));

  console.log('--- SEARCHING FOR TARGET JOB ---');
  const targetJob = db.prepare("SELECT * FROM jobs WHERE source_name LIKE '%2026-05-29%' OR source_name LIKE '%1.m4a%'").all();
  if (targetJob.length > 0) {
    for (const job of targetJob) {
      console.log(`Job Found: ${job.id} - ${job.source_name}`);
      console.log(`Status: ${job.status}`);
      console.log(`Transcript Length: ${job.transcript_text.length}`);
      console.log(`Transcript Snippet:\n${job.transcript_text.slice(0, 500)}...\n`);
      console.log(`Summary JSON:\n${job.summary_json}\n`);
    }
  } else {
    console.log('No matching job found!');
  }

} catch (err) {
  console.error(err);
} finally {
  db.close();
}
